internal/compiler/langtype.rs

```diff
@@ -777,12 +864,43 @@ pub struct Function {
     pub arg_names: Vec<SmolStr>,
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum StructName {
+    None,
+    /// When declared in .slint as  `struct Foo := { }`, then the name is "Foo"
+    User(SmolStr),
+    Native(NativeType),
+}
+
```

Comment:
Could also get something like

`impl<T: Into<NativeType>> From<T> for StructName`